### PR TITLE
Fix #450, correctly attaching children to folders from Sync

### LINKED ISSUE #450: Bookmarks are sometimes not synced in their corresponding folders.
Sometimes when you quickly add folders and bookmarks, syncing them on other device result in bookmarks being outside of folders

### PR DESCRIPTION
**Pull request by iccub**
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-ios/issues) for my issue if one did not already exist.
- [x] My patch or PR title has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!` (or `No Bug: <message>` if no relevant ticket)
- [ ] *Unit Tests* are updated to cover new or changed functionality
- [ ] User-facing strings use `NSLocalizableString()`
- [ ] New files have MPL-2.0 license header.

## Reviewer Checklist:

- [ ] PR is linked to an issue via [Zenhub](https://www.zenhub.com/extension).
- [ ] Issues are assigned to at least one epic.
- [ ] Issues include necessary QA labels:
  - [ ] `QA/(Yes|No)`
  - [ ] `release-notes/(include|exclude)`
  - [ ] `bug` / `enhancement`
- [ ] Necessary security reviews have taken place.
- [ ] Adequate test coverage exists to prevent regressions.
- [ ] Adaquate test plan exists for QA to validate (if applicable)

### CODE REVIEW

File: Data/models/Bookmark.swift [hunk 1]
```diff
@@ -196,17 +196,21 @@ public final class Bookmark: NSManagedObject, WebsitePresentable, Syncable, CRUD
...
             // Find all children and attach them
-            if let children = Bookmark.getNonFolderChildren(forFolderUUID: bk.syncUUID) {
-                
-                // TODO: Setup via bk.children property instead
-                children.forEach { $0.syncParentUUID = bk.syncParentUUID }
+            if let children = Bookmark.getChildren(forFolderUUID: bk.syncUUID, context: context) {
+                // Re-link all orphaned children
+                children.forEach {
+                    $0.syncParentUUID = bk.syncUUID
+                    $0.parentFolder = bk
```
Comment (jhreis) [inline Data/models/Bookmark.swift:215]: Definitely should add a comment here:

$0.syncParentUUID = bk.syncUUID
// The `setter` for `syncParentUUID` creates the parent/child relationship in CD, however in this specific instance
//  the objects have not been written to disk, so cannot be fetched on a different context and the relationship
//  will not be properly established. Manual attachment is necessary here during these batch additions.
$0.parentFolder = bk

File: Data/sync/Sync.swift [hunk 2]
```diff
@@ -448,52 +448,46 @@ extension Sync {
...
-                else { return }
-            
-            let clientRecord = recordType.coredataModelType?.get(syncUUIDs: [fetchedId], context: context)?.first as? Syncable
-            
-            var action = SyncActions(rawValue: fetchedRoot.action ?? -1)
-            if action == SyncActions.delete {
-                clientRecord?.remove(sendToSync: false)
-            } else if action == SyncActions.create {
+        context.perform {
+            for fetchedRoot in fetchedRecords {
```
Comment (jhreis) [inline Data/sync/Sync.swift:455]: Just to confirm, I do not see any logic changes inside this (just that it is wrapped inside a `.perform` is this a correct observation?

**Review by jhreis — CHANGES_REQUESTED** [review]

File: Data/sync/Sync.swift (hunk 2 above)
Comment (iccub) [reply Data/sync/Sync.swift:455]: Yes sir.

File: Data/models/Bookmark.swift (hunk 1 above)
Comment (iccub) [reply Data/models/Bookmark.swift:215]: Thanks, any suggestion to add comments are really appreciated by me :)

Comment (iccub) [reply Data/models/Bookmark.swift:215]: This is resolved btw

**Review by jhreis — APPROVED** [review]

**Pull request merged**